src/Core/Validation/FieldValidation/FieldValidator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\ValidationInterface;
+
+/**
+ * Abstract class that can be used as a validator for FormFields and DBFields
+ */
+abstract class FieldValidator implements ValidationInterface
+{
+    protected string $name;
+    protected mixed $value;
+
+    public function __construct(string $name, mixed $value)
+    {
+        $this->name = $name;
+        $this->value = $value;
+    }
+
+    /**
+     * Validate the value
+     */
+    public function validate(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        $validationResult = $this->validateValue($result);
+        if (!$validationResult->isValid()) {
+            $result->combineAnd($validationResult);
+        }
```

Comment:
Why are we passing a result in, but _also_ expecting a result to be returned? Should be one or the other.
Either:
```suggestion
        $result = $this->validateValue();
```
or:
```suggestion
        $result = ValidationResult::create();
        $this->validateValue($result);
```
Since `validateValue()` doesn't even take an argument, I recommend the former.

In either case, I can't see any reason for this to be split from the `validateValue()` method - what's the intention there?

Reply:
Has been updated